pym/bob/input.py

```diff
@@ -2373,6 +2373,10 @@ def __parse(self):
 
         # finally parse recipes
         if not os.path.isdir("recipes"):
+            self.__cache.close()
```

Comment:
Moving the test above self.__cache.open() would also result in closing a file never opened?